### PR TITLE
Add interface to xfer/collector.go, and some more tests

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

**Comment by tomwilkie** [comment]
This is another change broken out of #123

**Comment by rade** [comment]
This does more than the description suggests. In particular note the type change for the `quit` field, and the introduction of `peek`.

**Comment by peterbourgon** [comment]
- quit field change is to make it consistent with other Stoppable components in Scope
- peek is used only for tests, we could add a comment to that effect, but since it's not exported and only called in 1 place I believe it's pretty straightforward

File: xfer/collector.go [hunk 1]
```diff
@@ -17,51 +17,61 @@ const (
 
 var (
 	// MaxBackoff is the maximum time between connect retries.
-	MaxBackoff = 2 * time.Minute // externally configurable.
+	// It's exported so it's externally configurable.
+	MaxBackoff = 2 * time.Minute
+
+	tick = time.Tick
```
Comment (rade) [inline xfer/collector.go]: why have this? It's used exactly once, and makes the call site just that little bit more obscure.

**Comment by peterbourgon** [comment]
LGTM

File: xfer/collector.go (hunk 1 above)
Comment (tomwilkie) [inline xfer/collector.go]: tick is extracted out for mocking in the tests.

Comment (rade) [inline xfer/collector.go]: ah! comment, perhaps?

**Comment by tomwilkie** [comment]
I've updated to account for comments.  Will merge when tests pass.

File: xfer/collector.go (hunk 1 above)
Comment (tomwilkie) [inline xfer/collector.go]: Added https://github.com/weaveworks/scope/pull/176/files#diff-2d26093b1b10de6776f9b567b831f282R23

**Pull request merged**